File: backend/Models/Book.cs
using System.Collections;

public class Book
{
    public int BookId { get; set; }
    public required string Title { get; set; }
    public required string Author { get; set; }
    public int? Pages {get; set;}
    public string? Genre { get; set; }
    public string? Description { get; set; }
    public ICollection<Character> Characters { get; set; } = new List<Character>();

}

File: frontend/src/app/services/book.ts
export interface Book {
  bookId: number;
  title: string;
  author: string; // im Backend NICHT nullable
  genre?: string;
  description?: string;
  pages?: number;
  characters: Character[];
}
export interface CreateBook {
  title: string;
  author?: string;
}

export interface Character {
  characterId: number;
  name: string;
  description?: string;
  profession?: string;
  species?: Species;
}

export interface Species {
  speciesId: number;
  name: string;
  description?: string;
}
